### ISSUE TITLE
Obtaining path to the .git from the repository instance

### ISSUE THREAD

**Issue by mitar** [go-git/go-git#405]
I use:

```golang
	repository, err := git.PlainOpenWithOptions(somePath], &git.PlainOpenOptions{
		DetectDotGit: true,
	})
```

which automatically determines where is `.git` inside the path. But I could not find a way to obtain this path from `repository` afterwards. Am I missing something? If not, I would like to ask that this is provided somehow.

**Comment by ghostwheel42** [comment]
Hi @mitar
For local repositories I'm using
```
worktree, err := repository.Worktree()
// error handling omitted
print(worktree.Filesystem.Join(worktree.Filesystem.Root(), git.GitDirName))
```

**Comment by mitar** [comment]
Yes, but this is fragile because it is not necessary that `.git` is inside worktree. (I have been using a similar workaround myself for now.)

**Comment by mitar** [comment]
This should be simple, but would be very useful.

**Comment by pjbgf** [comment]
@mitar thanks for opening this issue and chasing it up.

There is a more recent issue with [some context](https://github.com/go-git/go-git/issues/977#issuecomment-2000616982) around how we could go about implementing this feature. Overall, any implementation here would be file system specific (e.g. memory vs osfs).

I am closing this as a duplicate of #977 - unfortunately I saw that one last month before seeing the issue you created.
